### ISSUE TITLE
RE: Issue #52 Doing my best to help. (Restarting miners on GPU lost) 

### ISSUE THREAD

**Issue by cmbconcretefire** [Sniffdog/Sniffdogminer#88]
Here is what I can tell you: > In theory, if your GPU's are dropping out , there is something under the hood going on. Theoretically, GPUS's should NEVER drop out. However, they simply do. Period. 

I bitched, and bitched, and bitched (and bitched some more) at Nice Hash Miner Legacy people to PLEASE implement some sort of "fix" for lost GPU's. Somebody finally wrote a script that was implemented into NHML called "OnGpuLost." (and I will post it here) It too uses an nvidia tool. 

The way the "OnGpulost" works seems so simple and elegant and I can confirm...... IT WORKS. 

I'm not going to lie, I've switched back and forth between NHML, MPM, and other miners. But only NHML has the magic bullet "fix" I'm looking for, which is a way to "poll" the GPU's and jump start them if one (or more) is lost. READ THE DIRECTIONS. Between what the other fellow wrote and this, it should be really simple to implement. And NO, it will not "fix" the underlying problem, but at least your rig will keep running while you sleep instead of being completely dead for 8 hours. You can worry about a real fix after the fact. 

[OnGPUsLost.txt](https://github.com/Sniffdog/Sniffdogminer/files/1650224/OnGPUsLost.txt)

**Comment by Sniffdog** [comment]
Thanks... much more refreshing than someone worrying about what I'm doing...

**Comment by Sniffdog** [comment]
#52 joining issues

**Comment by Archdemon** [comment]
So reading this right this would restart the computer on option 1?

What if the gpu aint lost but the miner get stuck in cpu invalid memory error?

Been having awesome miner in the background fir its temperature rules (restart on temp below X)

**Comment by Sniffdog** [comment]
yeah I think Ethmonitor has that function too... the link is on Hashrefinery home page

**Comment by cmbconcretefire** [comment]
@Archdemon , yes, option 1 would restart the whole rig. It is IMHO that option 1 is not really acceptable, nor does it make any sense. At least in my situation. IMHO, option 2 is the only option that makes any sense. Both options though create a 'log file' which can then (somewhat) be used to help identify the underlying problems THAT SHOULD BE FIXED buy the user..........however, this script / patch / whatever is an effective and temporary band aid to keep your rigs limping along while you sleep.......so that you don't lose 8 hrs of mining, all because one GPU dropped out.
